spec/integration/validator_spec.rb

```diff
@@ -114,6 +116,136 @@
     end
   end
 
+  context 'with valid inventory' do
+    let(:inventory) do
+      {
+        'groups' => [
+          {
+            'name' => 'nix',
+            'targets' => [
+              'nix1-example.org',
+              'nix2-example.org'
+            ],
+            'facts' => {
+              'operatingsystem' => 'linux'
+            },
+            'features' => [
+              'puppet-agent'
+            ]
+          },
+          {
+            'name' => 'windows',
+            'targets' => [
+              {
+                'name' => 'win1',
+                'uri' => 'win1-example.org',
+                'transport' => 'winrm'
+              }
+            ],
+            'groups' => [
+              {
+                'name' => 'subgroup',
+                'targets' => [
+                  'win3-example.org',
+                  'win4-example.org'
+                ],
+                'config' => {
+                  'winrm' => {
+                    'ssl' => false
+                  }
+                }
+              }
+            ]
+          }
+        ],
+        'config' => {
+          'transport' => 'ssh',
+          'ssh' => {
+            'host-key-check' => false
+          },
+          'winrm' => {
+            'user' => 'Administrator',
+            'password' => 'bolt'
+          }
+        }
+      }
+    end
+
+    it 'does not error' do
+      expect { run_cli(command, project: project) }.not_to raise_error
+    end
+  end
+
+  context 'with invalid inventory' do
+    let(:inventory) do
+      {
+        'groups' => [
+          {
+            'targets' => [
+              'nix1-example.org',
+              100
+            ],
+            'facts' => [
+              'linux'
+            ],
+            'features' => {
+              'puppet-agent' => true
+            }
+          },
+          {
+            'name' => 'windows',
+            'targets' => [
+              {
+                'name' => 'win1',
+                'uri' => 'win1-example.org',
+                'transport' => 'winrm'
+              }
+            ],
+            'groups' => {
+              'name' => 'subgroup',
+              'targets' => [
+                'win3-example.org',
+                'win4-example.org'
+              ],
+              'config' => {
+                'winrm' => {
+                  'ssl' => false
+                }
+              }
+            }
+          }
+        ],
+        'config' => {
+          'transport' => 'foo',
+          'ssh' => {
```

Comment:
I think it'd be useful to add examples here of keys that are valid for other transports but not `ssh`, and a key that isn't valid for any transport.

Reply:
Sorry I don't see the updates? Once those are tested I think this is GtG!

Reply:
There's a net new test "with unknown inventory options" at the bottom of the file. Can't test for warnings when there are errors, since Bolt errors before issuing the warnings.

Reply:
Oh weird I get some warnings before errors:
```
lucy ~/githubs/bolt ((b8269dae...)) $ be bolt command run whoami -t localhost
Unknown option 'ssl' at 'inventory-config.ssh' at /home/lucy/.puppetlabs/etc/bolt/bolt-defaults.yaml.
Invalid configuration at /home/lucy/githubs/bolt/inventory.yaml:
  Value at 'groups.0.config.ssh.user' must be of type String
```
:shrug:  Thanks!

Reply:
Ya, that first one is for default config, which is validated before inventory. :)